src/Microsoft.Azure.WebJobs.Host/Extensions/ExceptionExtensions.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Threading;
+using Microsoft.Azure.WebJobs.Host;
+
+namespace System
+{
+    internal static class ExceptionExtensions
+    {
+        public static bool IsFatal(this Exception exception)
+        {
+            while (exception != null)
+            {
+                if ((exception as OutOfMemoryException != null && exception as InsufficientMemoryException == null) ||
```

Comment:
Yeah, I thought the same. I didn't touch it (other than formatting) after copying from the link that Mathew sent -- [http://vasters.com/archive/Are-You-Catching-Falling-Knives.html](http://vasters.com/archive/Are-You-Catching-Falling-Knives.html)

I'll swap all of this to use 'is'.

---
In reply to: [76674918](https://github.com/Azure/azure-webjobs-sdk/pull/799#discussion_r76674918) [](ancestors = 76674918)